src/esb.c

```diff
@@ -202,7 +203,7 @@ void esbInterruptHandler()
       }
 
       // If this packet is a retry, send the same ACK again
-      if (isRetry(pk)) {
+      if (pk->ack && isRetry(pk)) {
```

Comment:
Yes, I think that should fix it. Sorry, I didn't manage to actually test that so far, otherwise I would have updated the PR already.